apps/wallet/src/ui/app/components/menu/content/VerifyLedgerConnectionStatus.tsx

```diff
@@ -35,45 +37,51 @@ export function VerifyLedgerConnectionStatus({
     const [verificationStatus, setVerificationStatus] = useState(
         VerificationStatus.UNKNOWN
     );
-    const timeoutIdRef = useRef<number>();
+    const { mutate, isLoading } = useMutation({
+        mutationFn: async () => {
+            const suiLedgerClient = await connectToLedger();
+            return await suiLedgerClient.getPublicKey(derivationPath, true);
+        },
+        onSuccess: (publicKeyResult) => {
+            const publicKey = new Ed25519PublicKey(publicKeyResult.publicKey);
+            const suiAddress = publicKey.toSuiAddress();
+
+            setVerificationStatus(
+                accountAddress === suiAddress
+                    ? VerificationStatus.VERIFIED
+                    : VerificationStatus.NOT_VERIFIED
+            );
+
+            window.setTimeout(() => {
```

Comment:
on mutation success the `verificationStatus` gets set, but there's a timeout that runs afterwards (potentially) to reset it to unknown instead of the result. should the unknown setting be in the error case?

Reply:
I thought more about this, yeah I think we should be setting the verification status to `NOT_VERIFIED` when there's a connection error (which is different than "Unknown" since that's the default "we don't know" state). Thanks for catching this!